### DISCUSSION TITLE
Incorrect padding - Wrong key

### DISCUSSION THREAD

**Discussion by Daythril** [noDRM/DeDRM_tools#209] [General]
DeDRM isn't working on a book I just bought.  Here's some text from the Calbre log:

> ApplicationPaletteChange event ignored
> Using calibre Qt style: True
> ApplicationPaletteChange event received
> calibre Debug log
> calibre 6.8  embedded-python: True
> Windows-10-10.0.19045-SP0 Windows ('64bit', 'WindowsPE')
> ('Windows', '10', '10.0.19045')
> Python 3.10.1
> Windows: ('10', '10.0.19045', 'SP0', 'Multiprocessor Free')
> Interface language: None
> Successfully initialized third party plugins: Gather KFX-ZIP (from KFX Input) (1, 49, 0) && DeDRM (10, 0, 3) && Package KFX (from KFX Input) (1, 49, 0) && KFX metadata reader (from KFX Input) (1, 49, 0) && KFX Input (1, 49, 0) && Set KFX metadata (from KFX Output) (1, 64, 0) && KFX Output (1, 64, 0) && Kindle Collections (1, 7, 29)
> calibre 6.8  embedded-python: True
> Windows-10-10.0.19045-SP0 Windows ('64bit', 'WindowsPE')
> ('Windows', '10', '10.0.19045')
> Python 3.10.1
> Windows: ('10', '10.0.19045', 'SP0', 'Multiprocessor Free')
> Interface language: None
> Successfully initialized third party plugins: Gather KFX-ZIP (from KFX Input) (1, 49, 0) && DeDRM (10, 0, 3) && Package KFX (from KFX Input) (1, 49, 0) && KFX metadata reader (from KFX Input) (1, 49, 0) && KFX Input (1, 49, 0) && Set KFX metadata (from KFX Output) (1, 64, 0) && KFX Output (1, 64, 0) && Kindle Collections (1, 7, 29)
> 

The problem _seems_ to be here:

> Decrypting KFX DRM voucher: voucher
> Traceback (most recent call last):
>   File "C:\Users\berkowitzm\AppData\Roaming\calibre\plugins\DeDRM.zip\kfxdedrm.py", line 105, in decrypt_voucher
>   File "C:\Users\berkowitzm\AppData\Roaming\calibre\plugins\DeDRM.zip\ion.py", line 886, in decryptvoucher
>   File "C:\Users\berkowitzm\AppData\Roaming\calibre\plugins\DeDRM.zip\ion.py", line 762, in pkcs7unpad
>   File "C:\Users\berkowitzm\AppData\Roaming\calibre\plugins\DeDRM.zip\ion.py", line 102, in _assert
> Exception: Incorrect padding - Wrong key
> 

The key is the same one I've been using forever; it has worked in the past.

Any ideas?

**Comment by ElleKayEm** [comment]
Looks like the newer KFX which cannot be decrypted.  Guessing Kindle for PC updated on you.  See this post for methods of getting books that can be decrypted and for preventing updates:  https://www.mobileread.com/forums/showthread.php?t=283371
